.github/actions/build_test_artifacts/action.yml

```diff
@@ -0,0 +1,24 @@
+name: Build Test Artifacts
+description: Common builder for test artifacts
+
+runs:
+  using: composite
+  steps:
+    - name: Cache cargo registry
+      uses: actions/cache@v2.1.7
+      with:
+        path: ~/.cargo/registry
+        key: ${{ runner.os }}-v2-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+    - name: Cache cargo index
+      uses: actions/cache@v2.1.7
+      with:
+        path: ~/.cargo/git
+        key: ${{ runner.os }}-v2-cargo-index-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Cashing is not part of Is this `.github/workflows/contracts-release.yml` workflow currently, but perhaps we could add it and reuse the action?